py/bitbox02/bitbox02/bitbox02/bitbox02.py

```diff
@@ -796,6 +785,59 @@ def eth_sign(self, transaction: bytes, keypath: Sequence[int], chain_id: int = 1
 
             return signature
 
+        if is_eip1559:
+            self._require_atleast(semver.VersionInfo(9, 16, 0))
+            (
+                decoded_chain_id,
```

Comment:
I added a check that this is the same as the passed chain_id param (seemed easier than forcing the arg to be `None` in case of eip1559).

Reply:
fixed